vulnerabilities/importers/npm.py

```diff
@@ -9,190 +9,140 @@
+            )
+        if cvss_vector and cvss_vector.startswith("CVSS:2.0/"):
+            severities.append(
+                VulnerabilitySeverity(
+                    system=CVSSV2,
+                    value=cvss_score,
+                )
             )
 
-        self._versions = NpmVersionAPI()
-        self.set_api(self.collect_packages())
-
-    def updated_advisories(self) -> Set[AdvisoryData]:
-        files = self._updated_files.union(self._added_files)
-        advisories = []
-        for f in files:
-            processed_data = self.process_file(f)
-            if processed_data:
-                advisories.extend(processed_data)
-        return self.batch_advisories(advisories)
-
-    def set_api(self, packages):
-        asyncio.run(self._versions.load_api(packages))
-
-    def collect_packages(self):
-        packages = set()
-        files = self._updated_files.union(self._added_files)
-        for f in files:
-            data = load_json(f)
-            packages.add(data["module_name"].strip())
+        advisory_reference = Reference(
+            url=f"https://github.com/nodejs/security-wg/blob/main/vuln/npm/{id}.json",
+            reference_id=id,
+            severities=severities,
+        )
 
-        return packages
+        for ref in data.get("references") or []:
+            references.append(
+                Reference(
+                    url=ref,
+                    severities=severities,
+                )
+            )
 
-    @property
-    def versions(self):  # quick hack to make it patchable
-        return self._versions
+        if advisory_reference not in references:
+            references.append(advisory_reference)
+
+        package_name = data.get("module_name")
+        affected_packages = []
+        if package_name:
+            affected_packages.append(self.get_affected_package(data, package_name))
+        advsisory_aliases = data.get("cves") or []
+
+        for alias in advsisory_aliases:
+            yield AdvisoryData(
+                summary=build_description(summary=summary, description=description),
+                references=references,
+                date_published=date_published,
+                affected_packages=affected_packages,
+                aliases=[alias],
+            )
 
-    def process_file(self, file) -> List[AdvisoryData]:
+    def get_affected_package(self, data, package_name):
+        vulnerable_range = data.get("vulnerable_versions") or ""
 
-        record = load_json(file)
-        advisories = []
-        package_name = record["module_name"].strip()
+        # https://github.com/nodejs/security-wg/blob/main/vuln/npm/213.json#L14
+        if vulnerable_range == "<=99.999.99999":
+            vulnerable_range = "*"
 
-        publish_date = parse(record["updated_at"])
-        publish_date = publish_date.replace(tzinfo=pytz.UTC)
+        affected_version_range = (
+            NpmVersionRange.from_native(vulnerable_range) if vulnerable_range else None
```

Comment:
I am not sure a ternary conditional help readability here... what about:
```
affected_version_range = None
if vulnerable_range:
    affected_version_range = NpmVersionRange.from_native(vulnerable_range)
```

Or even better, `NpmVersionRange.from_native` should handle a None or empty string and return a None in these cases